### PR TITLE
31755 Refactor Lorota Service

### PR DESCRIPTION
**Pull request by randomsync**
## Description of change

This PR updates the controllers and Lorota service to use the new clients. The changes are behind a feature flag: 'check_in_experience_services_refactor'. Testing and deployment of these changes will be handled in #31756

## Original issue(s)
department-of-veterans-affairs/va.gov-team#31755

## Things to know about this PR

- [x] rspec

### CODE REVIEW

**Review by LindseySaari — DISMISSED** [review]

**Comment by randomsync** [comment]
@dillo @kanchanasuriya surfacing this up. I think now we can coordinate merging this after NOK deployment. The refactored code will still be turned off and we can determine when to turn it on (with or without CHIP refactor that Dillo is working on).

File: modules/check_in/spec/request/v2/patient_check_ins_request_spec.rb [hunk 1]
```diff
@@ -63,6 +63,8 @@
       end
 
       it 'returns a success response' do
+        allow(Flipper).to receive(:enabled?)
```
Comment (kanchanasuriya) [inline modules/check_in/spec/request/v2/patient_check_ins_request_spec.rb:66]: As these statements being setup procedure for the spec, can we please have these defined under before block?

File: modules/check_in/spec/request/v2/patient_check_ins_request_spec.rb [hunk 2]
```diff
@@ -74,6 +76,23 @@
...
+
+      context 'with service_refactor feature flag on' do
+        it 'returns a success response' do
+          allow(Flipper).to receive(:enabled?)
+            .with('check_in_experience_services_refactor').and_return(true)
+          allow_any_instance_of(CheckIn::V2::Session).to receive(:redis_session_prefix).and_return('check_in_lorota_v2')
+          allow_any_instance_of(CheckIn::V2::Session).to receive(:jwt).and_return('jwt-123-1bc')
+          allow_any_instance_of(::V2::Lorota::Service).to receive(:check_in_data).and_return(resp)
+
+          Rails.cache.write(key, 'jwt-123-1bc', namespace: 'check-in-lorota-v2-cache')
```
Comment (kanchanasuriya) [inline modules/check_in/spec/request/v2/patient_check_ins_request_spec.rb:88]: Same as above, as these statements are for test setup, can we please have these defined under before block?

File: modules/check_in/spec/request/v2/sessions_request_spec.rb [hunk 3]
```diff
@@ -150,6 +150,8 @@
       end
 
       it 'returns a success response' do
+        allow(Flipper).to receive(:enabled?)
+          .with('check_in_experience_services_refactor').and_return(false)
         allow_any_instance_of(::V2::Lorota::Service).to receive(:token_with_permissions).and_return(service_resp)
         expect_any_instance_of(::V2::Lorota::Service).to receive(:token_with_permissions).once
```
Comment (kanchanasuriya) [inline modules/check_in/spec/request/v2/sessions_request_spec.rb:156]: Same as above comments

File: modules/check_in/spec/request/v2/sessions_request_spec.rb [hunk 4]
```diff
@@ -158,6 +160,20 @@
         expect(response.status).to eq(200)
         expect(JSON.parse(response.body)).to eq(service_resp[:permission_data])
       end
+
+      context 'with service_refactor feature flag on' do
+        it 'returns a success response' do
+          allow(Flipper).to receive(:enabled?)
+            .with('check_in_experience_services_refactor').and_return(true)
+          allow_any_instance_of(::V2::Lorota::Service).to receive(:token).and_return(service_resp)
+          expect_any_instance_of(::V2::Lorota::Service).to receive(:token).once
```
Comment (kanchanasuriya) [inline modules/check_in/spec/request/v2/sessions_request_spec.rb:169]: Same as above comments

**Review by kanchanasuriya — COMMENTED** [review]
Just general comment on moving test setup under before block

File: modules/check_in/spec/request/v2/patient_check_ins_request_spec.rb (hunk 1 above)
Comment (randomsync) [reply modules/check_in/spec/request/v2/patient_check_ins_request_spec.rb]: addressed in 6ea726dd366a3daf29ab012fc161ed7dd6a090f7

File: modules/check_in/spec/request/v2/patient_check_ins_request_spec.rb (hunk 2 above)
Comment (randomsync) [reply modules/check_in/spec/request/v2/patient_check_ins_request_spec.rb]: addressed in 6ea726dd366a3daf29ab012fc161ed7dd6a090f7

File: modules/check_in/spec/request/v2/sessions_request_spec.rb (hunk 3 above)
Comment (randomsync) [reply modules/check_in/spec/request/v2/sessions_request_spec.rb:156]: addressed in 6ea726dd366a3daf29ab012fc161ed7dd6a090f7

File: modules/check_in/spec/request/v2/sessions_request_spec.rb (hunk 4 above)
Comment (randomsync) [reply modules/check_in/spec/request/v2/sessions_request_spec.rb:169]: addressed in 6ea726dd366a3daf29ab012fc161ed7dd6a090f7

File: modules/check_in/spec/request/v2/sessions_request_spec.rb (hunk 3 above)
Comment (kanchanasuriya) [reply modules/check_in/spec/request/v2/sessions_request_spec.rb:156]: Could you please move these lines inside before block as well?

File: modules/check_in/spec/request/v2/sessions_request_spec.rb (hunk 4 above)
Comment (kanchanasuriya) [reply modules/check_in/spec/request/v2/sessions_request_spec.rb:169]: Could you please move these lines inside before block as well?

Comment (randomsync) [reply modules/check_in/spec/request/v2/sessions_request_spec.rb:173]: addressed in c9bbe5fbcc9d3bdba975960a72c9067777e42886

File: modules/check_in/spec/request/v2/sessions_request_spec.rb (hunk 3 above)
Comment (randomsync) [reply modules/check_in/spec/request/v2/sessions_request_spec.rb]: addressed in c9bbe5f

**Review by LindseySaari — APPROVED** [review]

**Review by kanchanasuriya — APPROVED** [review]
LGTM

**Review by dillo — APPROVED** [review]
LGTM

**Pull request merged**